crates/ruff/src/rules/flake8_builtins/rules/builtin_attribute_shadowing.rs

```diff
@@ -77,11 +77,12 @@ pub(crate) fn builtin_attribute_shadowing(
     if shadows_builtin(name, &checker.settings.flake8_builtins.builtins_ignorelist) {
         // Ignore shadowing within `TypedDict` definitions, since these are only accessible through
         // subscripting and not through attribute access.
-        if class_def
-            .bases
-            .iter()
-            .any(|base| checker.semantic().match_typing_expr(base, "TypedDict"))
-        {
+        if class_def.arguments.as_ref().is_some_and(|arguments| {
+            arguments
+                .args
+                .iter()
+                .any(|base| checker.semantic().match_typing_expr(base, "TypedDict"))
+        }) {
```

Comment:
Well this is a little less pretty.. would it make sense to add a helper to `ClassDef` to iterate over bases? (in a future pull request)

Reply:
Yeah we can add a `bases` method. I can even do it here.

Reply:
Done.